### PR TITLE
Fix gm s2 filters

### PR DESCRIPTION
**Pull request by NikitaGandhi**
filters is a tuple([("operation", <radius>), ...) not a dict anymore so I converted it to list object of operations and radius first and then get a max from radius list.

So not changing the implementation but fixed due to change in filters definition.

### CODE REVIEW

**Review by alexgleith — APPROVED** [review]

File: libs/algo/odc/algo/io.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def fuser(xx: xr.Dataset) -> xr.Dataset:
     pad = kw.pop("pad", None)
     if pad is None:
         if filters is not None:
-            pad = max(*filters.values())
+            pad = max(list(zip(*filters))[1])
```
Comment (alexgleith) [inline libs/algo/odc/algo/io.py:310]: Actually, what is the logic here?

Comment (NikitaGandhi) [inline libs/algo/odc/algo/io.py:310]: filters is a tuple not a dict anymore so I converted it to list object of operations and radius first and then get a max from radius list. 

So not changing the implementation but fixed due to change in filters definition.

Comment (alexgleith) [reply libs/algo/odc/algo/io.py:310]: Ok, I trust you. It's pretty hard to read, but if you're happy I'm happy!

Comment (NikitaGandhi) [reply libs/algo/odc/algo/io.py:310]: I know it is hart to iterate through tuple but this works :)

**Pull request merged**